components/content-service/pkg/layer/provider.go

```diff
@@ -465,15 +465,27 @@ func contentDescriptorToLayer(cdesc []byte) (*Layer, error) {
 	)
 }
 
+var prestophookScript = `#!/bin/bash
+cd ${GITPOD_REPO_ROOT}
+git config --global --add safe.directory ${GITPOD_REPO_ROOT}
+git status --porcelain=v2 --branch -uall > /.workspace/prestophookdata/git_status.txt
+git log --pretty='%h: %s' --branches --not --remotes > /.workspace/prestophookdata/git_log_1.txt
+git log --pretty=%H -n 1 > /.workspace/prestophookdata/git_log_2.txt
+`
+
 // version of this function for persistent volume claim feature
 // we cannot use /workspace folder as when mounting /workspace folder through PVC
 // it will mask anything that was in container layer, hence we are using /.workspace instead here
 func contentDescriptorToLayerPVC(cdesc []byte) (*Layer, error) {
-	return layerFromContent(
-		fileInLayer{&tar.Header{Typeflag: tar.TypeDir, Name: "/.workspace", Uid: initializer.GitpodUID, Gid: initializer.GitpodGID, Mode: 0755}, nil},
-		fileInLayer{&tar.Header{Typeflag: tar.TypeDir, Name: "/.workspace/.gitpod", Uid: initializer.GitpodUID, Gid: initializer.GitpodGID, Mode: 0755}, nil},
-		fileInLayer{&tar.Header{Typeflag: tar.TypeReg, Name: "/.workspace/.gitpod/content.json", Uid: initializer.GitpodUID, Gid: initializer.GitpodGID, Mode: 0755, Size: int64(len(cdesc))}, cdesc},
-	)
+	layers := []fileInLayer{
+		{&tar.Header{Typeflag: tar.TypeDir, Name: "/.workspace", Uid: initializer.GitpodUID, Gid: initializer.GitpodGID, Mode: 0755}, nil},
+		{&tar.Header{Typeflag: tar.TypeDir, Name: "/.workspace/.gitpod", Uid: initializer.GitpodUID, Gid: initializer.GitpodGID, Mode: 0755}, nil},
+		{&tar.Header{Typeflag: tar.TypeReg, Name: "/.supervisor/prestophook.sh", Uid: 0, Gid: 0, Mode: 0775, Size: int64(len(prestophookScript))}, []byte(prestophookScript)},
```

Comment:
Can you please explain why we are setting Uid and Gid to 0 but then permission to 0775?

Reply:
Uid/Gid to 0 to make sure no one else can edit\modify this file from inside the workspace (otherwise you can execute things from ring1).
775 to allow gitpod user to execute it as part of prestop hook.